### PR TITLE
HS-XXXXX Support missing MFA APIs

### PR DESCRIPTION
**Pull request by mozawa**
IAM MFA(Multi-Factor Authentication) APIs implementation.
- update params in create_virtual_mfa_device()
- add missing IAM:DeleteVirtualMFADevice API as delete_virtual_mfa_device()
- add missing IAM:ListVirtualMFADevices API as get_all_virtual_mfa_devices()

### CODE REVIEW

File: boto/iam/connection.py [hunk 1]
```diff
@@ -1722,7 +1722,7 @@ def get_credential_report(self):
         params = {}
         return self.get_response('GetCredentialReport', params)
 
-    def create_virtual_mfa_device(self, path, device_name):
+    def create_virtual_mfa_device(self, device_name, path='/'):
```
Comment (tpodowd) [inline boto/iam/connection.py]: What's the strong reason for switching these? Not saying we shouldn't but I guess Boto has it implemented this way by default so it seems like an odd change. Of course we are moving away from Boto to Boto3 perhaps so its not a huge deal, but normally, I wouldn't agree with this.

File: boto/iam/connection.py [hunk 2]
```diff
@@ -1743,6 +1743,56 @@ def create_virtual_mfa_device(self, path, device_name):
...
+        :param serial_number: The serial number which uniquely identifies
+            the MFA device. For virtual MFA devices, the serial number is
+            the same as the ARN.
+        """
+        params = {
+            'SerialNumber': serial_number
+        }
+        return self.get_response('DeleteVirtualMFADevice', params)
+
+    def get_all_virtual_mfa_devices(self, assignment_status=None, marker=None, max_items=None):
```
Comment (tpodowd) [inline boto/iam/connection.py:1762]: Probably should set assignment_status='Any' as the default here if it is going to be the default for None anyway?

**Review by tpodowd — COMMENTED** [review]
Have a read of my comments and let me know what you think.

File: boto/iam/connection.py (hunk 1 above)
Comment (mozawa) [reply boto/iam/connection.py]: Ok. The caller(iam.py) has 'path' variable default as '/' and calls this API with that. So this change was not necessary.
Back to the original one. https://github.com/cloudian/boto/commit/6d7e94522d14d710fddd350d226f257ca98913c3

File: boto/iam/connection.py (hunk 2 above)
Comment (mozawa) [reply boto/iam/connection.py:1762]: As 'AssignmentStatus' request param is optional, we want to test:
- No 'AssignmentStatus' request param in the request to see if Server will return the list of any status of virtual MFAs.
- 'AssignmentStatus' request param with 'Assigned/Unassigned/Any/other words, e.g. hoge' in the request to see if Sever will behave the corresponding response.

File: boto/iam/connection.py [hunk 3]
```diff
@@ -1743,6 +1743,56 @@ def create_virtual_mfa_device(self, path, device_name):
...
+        """
+        Lists the virtual MFA devices defined in the AWS account
+        by assignment status. If you do not specify an assignment status,
+        the operation returns a list of all virtual MFA devices.
+        Assignment status can be 'Assigned', 'Unassigned', or 'Any'.
+
+        :type assignment_status: string
+        :param assignment_status: The status (Unassigned or Assigned) of
+             the devices to list. If you do not specify an AssignmentStatus,
+             the operation defaults to Any, which lists both assigned and
```
Comment (tpodowd) [inline boto/iam/connection.py]: Let's changed the doc here...

> If you do not specify an AssignmentStatus, the operation lists both assigned and unassigned virtual MFA devices. You can also explicitly 'Any' which returns the same thing.

Comment (mozawa) [reply boto/iam/connection.py]: Thanks. Updated here. https://github.com/cloudian/boto/pull/23/commits/b6364e193ecd5096cb190e5d7ce64adb89cf536b

**Review by tpodowd — APPROVED** [review]
Thanks!

**Pull request merged**